Add readme how-to documentation including deserializing, extending, and structure (#120)
ruma/ruma: ruma-events/README.md

## Documentation

ruma-events has [comprehensive documentation](https://docs.rs/ruma-events) available on docs.rs.

# How-To

## Deserializing

> Ruma offers the `EventJson` wrapper to enable passing around JSON text that is semi validated. This

Reviewer: ```suggestion
Ruma offers the `EventJson` wrapper to enable passing around JSON that is only partially validated. This
```

> is useful when [extending ruma](TODO_link). All event structs and enums implement `Serialize/Deserialize` so

Reviewer: ```suggestion
is useful when [extending Matrix](TODO_link). All event structs and enums implement `Serialize/Deserialize` so
```

Author: Great now I can't remember what my `TODO_link` was all about :laughing: :facepalm:

> EventJson is not strictly needed anymore (pre 0.22 this was not the case).

Reviewer: This may be a bit misleading, for homeserver that want to support federation, it is strictly required to have a mechanism like that.

```rust
let json = r#"{ "type": "imagine a full event", "content": {...} }"#;

let deser = serde_json::from_str::<EventJson<AnyRoomEvent>>()
    .unwrap() // the first Result from serde_json::from_str
    .deserialize() // this deserializes the event while still keeping the RawJson in EventJson
    .unwrap() // finally get to the AnyRoomEvent

// event content has a trait function that allows for validated deserialization
let content = r#"{ "membership": "imagine member event content", ... }"#;
let deser = EventContent::from_parts("m.room.member", serde_json::value::to_raw_value(content)).unwrap();
// There is also a method to convert the content to the redacted version according to the Matrix spec
// The only field left is `membership`
let redacted = deser.redact();

> ```

Reviewer: If we want to talk about `EventJson`, its original motivation of not failing deserialization of the `/sync` response when individual events are invalid should proabably be mentioned. Also, I think this code example doesn't show what is described above (deserializing parts of a JSON payload and leading others in JSON form, but with the attached generic type that they're expected to deserialize to). We should probably be documenting `EventJson` separately from `ruma_events` altogether, as it's probably not going to stay there anyway (#32).

Author: I was more going for an answer to the commonly asked question "how do I deserialize this..." but, adding something specifically for `EventJson` would make sense as I skimmed over it and it is important for the reasons you outline. Having its own section where ever we decide to put this would be great.

Author: Also now that `serde_json::from_whatever::<Any*Event>(json)` works it may be more important to emphasize `EventJson`. That it is a "non-lossy" way to pass the events around

> ## Creating Events to Extend Ruma

Reviewer: I love this section! ♥️

Author: If you're talking about the `Creating Events to Extend Ruma` I know it's AWSOME how little code it takes to add your own event :partying_face:! Really good API we (you mostly) have come up with.

> For our example we will create a reaction message event. This can be used with ruma's event structs, for

Reviewer: ```suggestion
For our example we will create a reaction message event. This can be used with Rumas event structs, for
```
